### PR TITLE
Classes

### PR DESCRIPTION
**Pull request by badger-99**
This PR will introduce the following to the main dev branch:
- `Person` Superclass
- `Student` Subclass that is derived from the `Person` class.
- `Teacher` Subclass that is derived from the `Person` class.
- Project documentation.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def can_use_services?
+    true if of_age
+  end
+
+  private
+
+  def of_age?
+    true if @age >= 18 || @parent_permission
```
Comment (Ibizugbe) [inline person.rb]: Nice job so far😉. Just as required, the of_age method returns true if `@age` is greater then or equal to 18. and not if the user has permission from his/her parents😉. kindly adjust that to meet the requirements.
image

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..80_000)
+    @name = name
+    @age = age
+    @parent_permissioin = parent_permission
+  end
+
+  def can_use_services?
+    true if of_age
```
Comment (Ibizugbe) [inline person.rb]: Just as I pointed out earlier, the `can_use_service?` should now return true if the user is of age or has parents' permission😉. Kindly make adjustments here too👍🏼.

image

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom)
+    super()
+    @classroom = classroom
+  end
```
Comment (Ibizugbe) [inline student.rb]: Nice job implementing the student class to inherit from the person class😉. But just as required, the student class should extend the parent's (Person) constructor (name, age, and parent _permission)   by adding classroom to it😉. You can see the suggestions below and make the changes as required👍🏼.
image


```suggestion
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end

```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super()
+    @specialization = specialization
+  end
```
Comment (Ibizugbe) [inline teacher.rb]: Same thing applies here as stated above, the Constructor extends parent's constructor by adding @specialization and a parameter for it. Kindly take a look at how it was implemented above and use it as a guideline to implement the changes here👍🏼.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @badger-99 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] There is an individual `.rb` file for each class (`Person`, `Student`, and `Teacher`)👍🏼.
- [x] Linters are passing👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by badger-99** [comment]
@Ibizugbe Thank you so much for pointing out the errors in my code. The feedback you've given has been very constructive!

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @badger-99   👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉


## Highlight 👍 
✅ Made suggested changes


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**